### PR TITLE
send/test invocation form field bugfix

### PR DESCRIPTION
**Pull request by phetter**
**Debugging and Testing**

  - Variadic Args
    - Some form fields were not allowing input. This includes all fields on test invoke and the argument and amount fields on send invoke. 
      - SOLVED: use defaultValue instead of value on react Input component
    - Some form fields were not submitting values. 
      - SOLVED: implementation of Input component resulted in the component not being called properly.

### CODE REVIEW

File: src/app/components/common/form/InputField/index.js [hunk 1]
```diff
@@ -69,7 +69,7 @@ const InputField = ({
 
 InputField.propTypes = {
   onChangeHandler: PropTypes.func,
-  value: PropTypes.string.isRequired,
+  value: PropTypes.string,
```
Comment (DalderupMaurice) [inline src/app/components/common/form/InputField/index.js]: propTypes should be either required or have a default value

File: src/app/containers/SendInvoke/index.js [hunk 2]
```diff
@@ -27,13 +29,15 @@ class SendInvoke extends Component {
     this.setState({
       [key]: e.target.value,
     })
+    logDeep('key: ', key + ' ' + e.target.value)
```
Comment (DalderupMaurice) [inline src/app/containers/SendInvoke/index.js]: could we remove logging everywhere?

File: src/app/containers/SendInvoke/index.js [hunk 3]
```diff
@@ -27,13 +29,15 @@ class SendInvoke extends Component {
...
   }
 
   _handleArgChange = (id, e) => {
-    const myArgs = this.state.args
+    var myArgs = this.state.args
```
Comment (DalderupMaurice) [inline src/app/containers/SendInvoke/index.js]: Why the need for var here? (and other places)

File: src/app/containers/SendInvoke/index.js [hunk 4]
```diff
@@ -46,17 +50,18 @@ class SendInvoke extends Component {
     this.setState({ args: this.state.args.filter((s, idx) => id !== idx) })
   }
 
-  handleSubmit = event => {
+  _handleSubmit = event => {
```
Comment (DalderupMaurice) [inline src/app/containers/SendInvoke/index.js]: for consistency reasons, It'd be good practice to name all handlers with a `_` or without. Currently it's mixed

File: src/app/containers/SendInvoke/index.js [hunk 5]
```diff
@@ -46,17 +50,18 @@ class SendInvoke extends Component {
...
+    const { scriptHash, operation, args } = this.state
 
     this.setState({
       loading: true,
       errorMsg: '',
       txid: '',
     })
 
-    if (!this.state.scriptHash || !this.state.operation || !this.state.amount) {
+    if (!scriptHash || !operation || !assetAmount) {
```
Comment (DalderupMaurice) [inline src/app/containers/SendInvoke/index.js]: where does `assetAmount` come from?

File: src/app/containers/SendInvoke/index.js [hunk 6]
```diff
@@ -134,13 +139,14 @@ class SendInvoke extends Component {
...
             id='assetAmount'
-            onChange={ this._handleInputChange }
+            onChangeHandler={ this._handleInputChange }
           />
           <Select
             cssOnly
             label='Asset'
             value={ this.state.assetType }
-            onChange={ e => {
+            onChangeHandler={ e => {
```
Comment (DalderupMaurice) [inline src/app/containers/SendInvoke/index.js]: remove logging (+ other places)

File: src/app/containers/TestInvoke/index.js [hunk 7]
```diff
@@ -66,7 +72,7 @@ class TestInvoke extends Component {
     if (!scriptHash || !operation) {
       this.setState({
         loading: false,
-        errorMsg: 'Error! Script hash and operation are both required!',
+        errorMsg: 'Error! Script hash and operation are both required now!!',
```
Comment (DalderupMaurice) [inline src/app/containers/TestInvoke/index.js]: now?

**Comment by phetter** [comment]
Regarding your comments, I'm doing the plan above @DalderupMaurice . So far I have no other committment from anyone else so I'm updating the dev branch liberally. If anyone else is working on this please let me know so I can adjust my course.
Many of the things you are commenting on are strictly temporary to reach the goals outlined.

File: src/app/containers/SendInvoke/index.js (hunk 3 above)
Comment (phetter) [reply src/app/containers/SendInvoke/index.js]: This is one that slipped by.

**Comment by phetter** [comment]
Test failed due to snapshot @DalderupMaurice

**Pull request merged**